src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -1311,6 +1311,44 @@ export class ResultsViewPageStore {
         },
     });
 
+
+    readonly remoteNgchmUrl = remoteData<string>({
+        await:()=>[this.studyIds],
+        invoke:async()=>{
+            var result = '';
+
+            if (this.studyIds.result!.length === 1) {
+
+                const queryData = {
+                    studyid:this.studyIds.result![0],
+                    format:"json"
+                };
+
+                const urlResponse = await request.get("https://bioinformatics.mdanderson.org/study2url")
+                                          .query(queryData) as any;
+
+                if (urlResponse.body.fileContent) {
+                    const parsedUrlResponse = JSON.parse(urlResponse.body.fileContent.trimEnd()) as any;
+
+                    if (parsedUrlResponse.length >= 1) {
+                        //result = "https://bioinformatics.mdanderson.org/study2url?" + parsedUrlResponse[0];
+                        // This is faked out for now.  study2url needs mods to include site url
+                        result = "https://bioinformatics.mdanderson.org/TCGA/NGCHMPortal?" + parsedUrlResponse[0];
+                    }
+                }
+            }
+
+            return Promise.resolve (result);
+        }
+    });
+
+    readonly isThereRemoteNGCHM = remoteData<boolean>({
```

Comment:
rather than creating a wrapper promise to return boolean, just put what's in invoke logic down in ngchmButtonActive getter.   We're trying to be more judicious about creating lots of wrapper remoteData (mobxpromise), by which i mean, remoteData instances that only exist to process other remoteData and don't actually make any async calls.

Reply:
Did as you suggested, and it seems to work, though it's not obvious to me why this doesn't cause a race condition other than ngchmButtonActive just happens to never be called before the remoteData is already resolved.  (Still in RC for now.)